src/main/java/booktopiahub/controller/CategoryController.java

```diff
@@ -0,0 +1,59 @@
+package booktopiahub.controller;
+
+import booktopiahub.dto.category.CategoryDto;
+import booktopiahub.dto.category.CreateCategoryRequestDto;
+import booktopiahub.service.category.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @GetMapping
+    @Operation(summary = "Get all categories",
+            description = "Get a list of all available categories")
+    public List<CategoryDto> findAll(Pageable pageable) {
```

Comment:
I don't think swagger will work correctly with pagable without this annotation.
Please recheck and correct in all controllers if I'm correct
```suggestion
    public List<CategoryDto> findAll(@ParameterObject Pageable pageable) {
```

Reply:
Updated

Reply:
Also added requirements according roles